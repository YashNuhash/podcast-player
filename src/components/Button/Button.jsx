import React from 'react'
import styles from './Button.module.scss'

export const Button = ({ children, inverted, ...otherProps }) => (
  <button
    type='button'
    className={`${styles.Button} ${inverted ? styles.Inverted : ''}`}
    {...otherProps}
  >
    {children}
  </button>
)

export default Button
