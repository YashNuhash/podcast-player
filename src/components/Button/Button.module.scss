@import '../../utils';

.Button {
    text-align: center;
    padding: 1rem 2rem;
    background-color: $primary;
    color: $offset;
    border: none;
    border-radius: 100px;
    font-weight: 600;
    cursor: pointer;
}

.Inverted {
    background-color: $offset;
    color: $primary;
}
