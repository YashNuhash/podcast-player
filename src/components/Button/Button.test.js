import React from 'react'
import { shallow } from 'enzyme'
import { Button } from './Button'

describe('Button', () => {
    it('should render default version', () => {
        expect(shallow(<Button>Hello</Button>)).toMatchSnapshot()
    })
    it('should render inverted version', () => {
        expect(shallow(<Button inverted>Hello</Button>)).toMatchSnapshot()
    })
})
