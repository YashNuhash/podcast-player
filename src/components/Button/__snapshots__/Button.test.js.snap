// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render default version 1`] = `
<button
  className="Button "
  type="button"
>
  Hello
</button>
`;

exports[`Button should render inverted version 1`] = `
<button
  className="Button Inverted"
  type="button"
>
  Hello
</button>
`;

exports[`should render Button component 1`] = `
<button
  className="Button "
  type="button"
>
  Hello
</button>
`;
